opam-ci-check/bin/main.ml

```diff
@@ -140,35 +194,63 @@ let pkg_term =
   let info = Arg.info [] ~doc:"Package name + version" in
   Arg.required (Arg.pos 0 (Arg.some Arg.string) None info)
 
-let changed_pkgs_term =
-  let info =
-    Arg.info
-      [ "c"; "changed-packages" ]
-      ~doc:"List of changed package name + version"
+let package_specs_term =
+  let package_spec_conv =
+    let parse_package_spec s =
+      try
+        let parts = String.split_on_char ':' s in
+        let name_version = List.hd parts in
+        let pkg = OpamPackage.of_string name_version in
+        let attrs =
+          match List.tl parts |> String.concat ":" with
+          | "" -> []
```

Comment:
IIUC, given the docs on the spec format, isn't this only valid if the split list is either a singletion or a two item list? It dosen't seem like more than one `:` should be able to occur in a term. So I think this maybe could be given better error handling with something like

```suggestion
        let parts = String.split_on_char ':' s in
        let* name_version, attrs = match parts with
          | [n] -> Ok (n, [])
          | [n; specs] -> Ok (n, String.split_on_char ',' specs)
          | _ -> Error ...
        in
        let pkg = OpamPackage.of_string name_version in
        let attrs = (* parsing apprs, with error handling, etc. *)
```